### PR TITLE
Bug 1244629 - Migrate remaining mozL10n.get calls to async in System

### PR DESCRIPTION
**Pull request by zbraniecki**
https://bugzilla.mozilla.org/show_bug.cgi?id=1244629

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0014e05c18286ed8b530d06cb3485d7f62b18e9e

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5b1d83fbff771c3d59396d8c6f1f6af39d61219f

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3ac95e93eb51cb46c6022e5f3a0f962b8c646268

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=26f9731ef8fc6c63de0806212a2c21a2987b5e85

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=30e2aac0444ab165bb857ba1bc62cd31daeddf6c

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=84c4fccb452c48f00e5c250787d4abc3f58ddd46

File: apps/system/js/accessibility.js [hunk 1]
```diff
@@ -809,26 +803,23 @@
         this.speech.resume();
       }
 
-      var sentence = this.buildUtterance(aData);
-      if (!sentence) {
-        return Promise.resolve();
-      }
+      return this.buildUtterance(aData).then(sentence => {
+        return new Promise((resolve, reject) => {
```
Comment (yzen) [inline apps/system/js/accessibility.js]: Nit: reject is not used.

File: apps/system/js/accessibility.js [hunk 2]
```diff
@@ -809,26 +803,23 @@
         this.speech.resume();
       }
 
-      var sentence = this.buildUtterance(aData);
-      if (!sentence) {
-        return Promise.resolve();
-      }
+      return this.buildUtterance(aData).then(sentence => {
```
Comment (yzen) [inline apps/system/js/accessibility.js]: Can we also add:
```
if (!sentence) {
  return Promise.resolve();
}
```

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=25764e7ea5dd2f651a7c4912b260d91750acc0c8

**Pull request merged**